pkg/controllers/terminating/v1alpha1/suite_test.go

```diff
@@ -75,21 +75,24 @@ var _ = Describe("Reallocation", func() {
 	})
 
 	Context("Reconciliation", func() {
-		It("should terminate nodes marked terminable", func() {
+		It("should terminate nodes marked drained", func() {
 			node := test.NodeWith(test.NodeOptions{
 				Labels: map[string]string{
 					v1alpha1.ProvisionerNameLabelKey:      provisioner.Name,
 					v1alpha1.ProvisionerNamespaceLabelKey: provisioner.Namespace,
-					v1alpha1.ProvisionerPhaseLabel:        v1alpha1.ProvisionerTerminablePhase,
 				},
 				Annotations: map[string]string{
 					v1alpha1.ProvisionerTTLKey: time.Now().Add(time.Duration(-100) * time.Second).Format(time.RFC3339),
 				},
 			})
 			ExpectCreatedWithStatus(env.Client, node)
 
-			ExpectCreated(env.Client, provisioner)
+			persisted := node.DeepCopy()
+			node.Labels[v1alpha1.ProvisionerPhaseLabel] = v1alpha1.ProvisionerDrainedPhase
+			Eventually(Expect(env.Client.Patch(ctx, node, client.MergeFrom(persisted))).To(Succeed()))
 
+			// Let some time go by
```

Comment:
Is this comment accurate?

Reply:
Creating the provisioner gives us a little more leeway to reduce this race condition. It's a little hacky, but will change when we have more control over the termination process, for instance `do-not-evict` pods.

Reply:
Let's talk.